src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -600,6 +610,34 @@ public boolean canProvision(@NonNull Cloud.CloudState state) {
     public PodTemplate getTemplate(@CheckForNull Label label) {
         return PodTemplateUtils.getTemplateByLabel(label, getAllTemplates());
     }
+    
+    @CheckForNull
+    public PodTemplate getTemplate(@NonNull String id) {
+        return getTemplateById(id);
+    }
+
+    // TODO: Remove when JENKINS-71737 is fixed
+    @POST
+    public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {
+        
+        Jenkins j = Jenkins.get();
+        j.checkPermission(Jenkins.ADMINISTER);
+        Cloud cloud = j.getCloud(this.name);
+        if (cloud == null) {
+            return HttpResponses.notFound();
+        }
+        KubernetesCloud result = (KubernetesCloud) cloud.reconfigure(req, req.getSubmittedForm());
+        String proposedName = result.name;
+        if (!proposedName.equals(this.name)
+                && j.getCloud(proposedName) != null) {
+            throw new Descriptor.FormException(Messages.cloudAlreadyExists(proposedName), "name");
+        }
+        result.templates = this.templates;
+        j.clouds.replace(this, result);
+        j.save();
+        // take the user back to the cloud top page.
+        return FormApply.success("../..");
+    }
```

Comment:
Even if this starts from a good intention, I think it should be split to a separate PR as it is unrelated to the intent of this PR.